Sources/Sentry/SentryANRTracker.m

```diff
@@ -47,60 +48,33 @@ - (instancetype)initWithTimeoutInterval:(NSTimeInterval)timeoutInterval
 - (void)detectANRs
 {
     NSThread.currentThread.name = @"io.sentry.app-hang-tracker";
-
     self.thread = NSThread.currentThread;
 
-    BOOL wasPreviousANR = NO;
+    __block NSInteger ticksSinceUiUpdate = 0;
+    NSInteger reportTreshold = 5;
+    double sleepIntervalMs = self.timeoutInterval / reportTreshold;
 
     while (![self.thread isCancelled]) {
+        ticksSinceUiUpdate++;
 
-        NSDate *blockDeadline =
-            [[self.currentDate date] dateByAddingTimeInterval:self.timeoutInterval];
-
-        __block BOOL blockExecutedOnMainThread = NO;
-        [self.dispatchQueueWrapper dispatchOnMainQueue:^{ blockExecutedOnMainThread = YES; }];
+        [self.dispatchQueueWrapper dispatchOnMainQueue:^{
+            ticksSinceUiUpdate = 0;
+            self->reported = NO;
+        }];
 
-        [self.threadWrapper sleepForTimeInterval:self.timeoutInterval];
+        [self.threadWrapper sleepForTimeInterval:sleepIntervalMs];
 
-        if (blockExecutedOnMainThread) {
-            if (wasPreviousANR) {
-                [SentryLog logWithMessage:@"ANR stopped." andLevel:kSentryLevelWarning];
-                [self ANRStopped];
+        if (ticksSinceUiUpdate >= reportTreshold && !reported) {
+            if (![self.crashWrapper isApplicationInForeground]) {
+                [SentryLog logWithMessage:@"Ignoring ANR because the app is in the background"
+                                 andLevel:kSentryLevelDebug];
+                continue;
```

Comment:
Will this spam the logs as long as the app remains backgrounded? If so we could wrap just the log statement in an `if` that checks a boolean like `didLogBgWarning`.

Reply:
I moved the `self->reported = YES` line up, so now it should only log once.